### PR TITLE
Queded dma test1

### PR DESCRIPTION
**Pull request by kikass13**

### CODE REVIEW

**Comment by kikass13** [comment]
important mdom internal files:


* [modm/src/modm/platform/spi/stm32/spi_master_queued_dma.hpp.in](https://github.com/DroidDrive/modm/blob/queded_dma_test1/src/modm/platform/spi/stm32/spi_master_queued_dma.hpp.in)
* [modm/src/modm/platform/spi/stm32/spi_master_queued_dma_impl.hpp.in](https://github.com/DroidDrive/modm/blob/queded_dma_test1/src/modm/platform/spi/stm32/spi_master_queued_dma_impl.hpp.in)
* [modm/src/architecture/interface/spi.hpp](https://github.com/DroidDrive/modm/blob/queded_dma_test1/src/modm/architecture/interface/spi.hpp)

important modm driver files:
 * [modm/src/driver/can/mcp2515_dmaint.hpp](https://github.com/DroidDrive/modm/blob/queded_dma_test1/src/modm/driver/can/mcp2515_dmaint.hpp)
 * [modm/src/driver/can/mcp2515_dmaint_impl.hpp](https://github.com/DroidDrive/modm/blob/queded_dma_test1/src/modm/driver/can/mcp2515_dmaint_impl.hpp)
 *  [modm/src/driver/inertial/adis16470_dmaint.hpp](https://github.com/DroidDrive/modm/blob/queded_dma_test1/src/modm/driver/inertial/adis16470_dmaint.hpp)
 *  [modm/src/driver/inertial/adis16470_dmaint_impl.hpp](https://github.com/DroidDrive/modm/blob/queded_dma_test1/src/modm/driver/inertial/adis16470_dmaint_impl.hpp)

test case is implemented here:

* https://github.com/DroidDrive/low-level-compute/tree/adis_queued_dma_fixes

in:
 * https://github.com/DroidDrive/low-level-compute/blob/adis_queued_dma_fixes/app/src/main.cpp
 * https://github.com/DroidDrive/low-level-compute/blob/adis_queued_dma_fixes/app/include/tasks/ImuDriver.hpp

**Comment by kikass13** [comment]
@marhmm-ducktrain  we can do a live (max 30 min) review if you want :)

File: src/modm/architecture/interface/spi.hpp [hunk 1]
```diff
@@ -45,6 +49,92 @@ struct Spi
...
+		post{[wrapper](){
+			if constexpr (!std::is_void<typename decltype(wrapper)::CS>::value){
+				switch(wrapper.behavior){
+					case ChipSelect::NONE: break;
+					case ChipSelect::SET: decltype(wrapper)::CS::set(); break;
+					case ChipSelect::TOGGLE: decltype(wrapper)::CS::set(); break;
+					default: break;
+				};
+			}
+		}}{}
```
Comment (marhmm-ducktrain) [inline src/modm/architecture/interface/spi.hpp:99]: Klammer zu viel?

Comment (kikass13) [reply src/modm/architecture/interface/spi.hpp:99]: nö passt, ist nur schlecht eingerückt weil ich in den modm scheiß nicht immer clangformat nutzen kann

**Comment by kikass13** [comment]
i read somewhere, that there could be a problem with dma complete interrupts when the interrupt priorities are badly configured ... 

which seems to NOT be the case here

from:

* modm/platform/DmaController<id>/Channel

```
	static void
		enableInterruptVector(uint32_t priority = 1)
		{
			NVIC_SetPriority(DmaBase::Nvic<ID>::DmaIrqs[uint32_t(ChannelID)], priority);
			NVIC_EnableIRQ(DmaBase::Nvic<ID>::DmaIrqs[uint32_t(ChannelID)]);
		}
```

from:

* modm/platform/extint/stm32/ext.hpp

```
connect(Trigger trigger, Handler&& handler, uint8_t priority=15)
	{
		...
		enableVector<Pin>(priority);
                ...
	}
```

**Comment by kikass13** [comment]
full frequency (YIELD_TASK) can send looks like this:

 * canbusload can0@1000000 -t -b -c -i
 
```
canbusload 2022-09-29 11:46:12 (ignore bitstuffing)
 can0@1000000  9500  940500 304000  94% |XXXXXXXXXXXXXXXXXX..|
```


 * candump (delta time between messages) looks like this

```
(000.001004)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000103)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000901)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000097)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.001009)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000096)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000899)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000096)  can0  00001234   [4]  AA FF 00 AA
 (000.000001)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
 (000.000000)  can0  00001234   [4]  AA FF 00 AA
```

**Comment by kikass13** [comment]
image

**Pull request merged**